packages/core/lib/testing/testrunner.js

```diff
@@ -175,7 +175,11 @@ TestRunner.prototype.endTest = async function(mocha) {
     fromBlock: this.currentTestStartBlock.toNumber()
   });
 
-  if (logs.length === 0) {
+  const userDefinedEventLogs = logs.filter(log => {
+    return !log.decodings.length || log.decodings[0].abi.name !== "TestEvent";
```

Comment:
```suggestion
    return log.decodings.every(decoding => decoding.abi.name !== "TestEvent")
```

(I guess this uses `every` rather than `some` now that I write it out, but this is the sort of thing I was talking about; thought maybe I should illustrate...)

Reply:
Ok so this ensures that none of the decodings contain `TestEvent`s. Will this give us the behavior we want? Like will there ever be any weirdness where we want to keep certain `log` objects that may contain `TestEvent`s and other values for `decoding.abi.name`?

Reply:
Well, I mean, I guess I can't *guarantee* that won't happen, but it would require a keccak256 collision so it seems pretty unlikely?  (Either that, or someone defining their own `TestEvent`s, in which case, well...)  I think hash collisions like that are in the realm of "things we can assume won't happen".